resnet.c

```diff
@@ -0,0 +1,345 @@
+/* Copyright (C) 2018, Oracle and/or its affiliates. All rights reserved
+ *
+ * This file is part of libresource.
+ *
+ * libresource is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * libresource is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with libresource. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _RESOURCE_H
+#include "resource.h"
+#endif
+#include <string.h>
+#include <unistd.h>
+#include <malloc.h>
+#include <errno.h>
+#include <ctype.h>
+#include <net/if.h>
+#include "resnet.h"
+#include "resource_impl.h"
+
+static inline int procnetdev_ver(char *buf)
+{
+	if (strstr(buf, "compressed"))
+		return 3;
+	if (strstr(buf, "bytes"))
+		return 2;
+	return 1;
+}
+
+/* Fetch interface name from interface information line */
+static inline char *get_ifname(char *name, char *p)
+{
+	char *d;
+	char *dname;
+
+	while (isspace(*p))
+		p++;
+
+	while (*p) {
+		if (isspace(*p))
+			break;
+		if (*p == ':') {
+			d = p, dname = name;
+			*name++ = *p++;
+			while (isdigit(*p))
+				*name++ = *p++;
+			if (*p != ':') {
+				p = d;
+				name = dname;
+			}
+			if (*p == '\0')
+				return NULL;
+			p++;
+			break;
+		}
+		*name++ = *p++;
+	}
+	*name++ = '\0';
+	return p;
+}
+
+/* Scan network interface stat line for interface and populate all relevant
+ * information.
+ */
+static inline int scan_net_stat(char *buf, res_net_ifstat_t *i, int ver)
```

Comment:
Hm, that is a lot of parsing. Any reason why you wouldn't be going through `netlink`?

Reply:
I took inspiration from netstat code when I was working on it, however I agree that I can add more comments.
Regarding netlink, you mean to fetch both stat and interface name using netlink right ? I remeber I tried that too but did not see any performance improvement so just did what netstat was doing. However if netlink is the way to go in future I guess I should change this. Let me know if I need to change this to netlink.